sync/requests.go

```diff
@@ -114,14 +114,19 @@ func atxsAsItems(msg []byte) ([]Item, error) {
 }
 
 func txsAsItems(msg []byte) ([]Item, error) {
-	var txs []types.SerializableSignedTransaction
+	var txs []*types.Transaction
 	err := types.BytesToInterface(msg, &txs)
 	if err != nil || txs == nil {
 		return nil, err
 	}
 	items := make([]Item, len(txs))
 	for i := range txs {
-		items[i] = &txs[i]
+		err := txs[i].CalcAndSetOrigin()
+		if err != nil {
+			return nil, fmt.Errorf("failed to calc transaction origin (id: %s): %v",
+				txs[i].Id().ShortString(), err)
```

Comment:
need to discuss whether or not we should not continue iterating over all other items in case we fail.

Reply:
Sure, we can discuss.

The logic here is that if I requested a list of transactions from a peer and they gave me a syntactically invalid transaction then I'd rather not trust that peer anymore.

The other option, as I guess you suggest, is to remove this specific item from the list and take advantage of the valid items the peer did provide.

Both are valid options, IMO.

Since it's handling of an adversarial situation, I'm generally inclined to prefer minimizing the attacker's power over saving some time/bandwidth by using other data they provide.